core/dbt/contracts/graph/compiled.py

```diff
@@ -122,7 +122,17 @@ class CompiledSchemaTestNode(CompiledNode, HasTestMetadata):
     def same_config(self, other) -> bool:
         return (
             self.unrendered_config.get('severity') ==
-            other.unrendered_config.get('severity')
+            other.unrendered_config.get('severity') and
+            self.unrendered_config.get('where') ==
+            other.unrendered_config.get('where') and
+            self.unrendered_config.get('limit') ==
+            other.unrendered_config.get('limit') and
+            self.unrendered_config.get('fail_calc') ==
+            other.unrendered_config.get('fail_calc') and
+            self.unrendered_config.get('error_if') ==
+            other.unrendered_config.get('error_if') and
+            self.unrendered_config.get('warn_if') ==
+            other.unrendered_config.get('warn_if')
```

Comment:
Here's what I was doing over in #3316:
```python
def same_config(self, other) -> bool:
        comparisons = [
            self.unrendered_config.get(modifier) == other.unrendered_config.get(modifier) or (
                self.unrendered_config.get(modifier) is None and
                other.unrendered_config.get(modifier) is None
            )
            for modifier in ('severity', 'store_failures', ...)
        ]
        return all(comparisons)
```
I'm not sure if that's better or worse. Might not work at all. Checking for `None` does seem important though

Reply:
ya! was planning on going back to do this, love it.